types/service/database_service_configuration.go

```diff
@@ -14,6 +14,8 @@ const (
 	DatabaseServiceTypeStandard    = "standard"     // standard MySQL or PostgreSQL, supports TLS and password auth
 	DatabaseServiceTypeAwsRds      = "aws_rds"      // AWS RDS database, supports IAM and password auth
 	DatabaseServiceTypeGcpCloudSql = "gcp_cloudsql" // Google Cloud SQL database, supports IAM, TLS and password auth
+	DatabaseServiceTypeAzureSql    = "azure_sql"    // Azure SQL database, supports SQL authentication, azure password auth
+	DatabaseServiceTypeSqlServer   = "sqlserver"    // Microsoft SQL Server database, supports SQL authentication and kerberos auth
```

Comment:
I don't see how to do this, we have the following standard auth types (both username/password):
Kerberos          *UsernameAndPassword `json:"kerberos_configuration,omitempty"`
SqlAuthentication *UsernameAndPassword `json:"sql_authentication_configuration,omitempty"`

And if we add this to standard, than this only applies to sqlserver not mysql/postgres.

Reply:
We can add a check so that it is not set for anything other than dbtype sqlserver. 

Im fine with whatever -- i just think its inconsistent with what we have.